src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2011-Present, CauseCode Technologies Pvt Ltd, India.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are not permitted.
+ */
+package com.causecode.fileuploader.util.renewer
+
+import static com.google.common.base.Preconditions.checkArgument
+
+import com.causecode.fileuploader.CDNProvider
+import com.causecode.fileuploader.StorageException
+import com.causecode.fileuploader.UFile
+import com.causecode.fileuploader.UFileType
+import com.causecode.fileuploader.cdn.CDNFileUploader
+import com.causecode.fileuploader.util.Time
+import com.causecode.util.NucleusUtils
+import grails.util.Holders
+import groovy.util.logging.Slf4j
+
+/**
+ * This renewer, fetches UFiles of given CDNProvider and then renews its temporary urls.
+ *
+ * @author Milan Savaliya
+ * @author Hardik Modha
+ * @since 3.1.3
+ */
+@Slf4j
+@SuppressWarnings('FieldName') // Regex for final fields is incorrect in CodeNarc rules
+class DefaultUFileTemporaryUrlRenewer implements UFileTemporaryUrlRenewer {
+
+    private final CDNProvider cdnProvider
+    private final CDNFileUploader cdnFileUploader
+    private final int maxResultsInOneIteration
+    private final boolean forceAll
+
+    DefaultUFileTemporaryUrlRenewer(
+            CDNProvider cdnProvider,
+            CDNFileUploader cdnFileUploader,
+            boolean forceAll,
+            int maxResultsInOneIteration = 100) {
+
+        checkArgument(cdnProvider != null, 'CDNProvider can not be null')
+        checkArgument(cdnFileUploader != null, 'CDNFileUploader can not be null')
+        checkArgument(maxResultsInOneIteration > 0, 'maxResultsInOneIteration can\'t be negative or zero')
+
+        this.cdnProvider = cdnProvider
+        this.cdnFileUploader = cdnFileUploader
+        this.forceAll = forceAll
+        this.maxResultsInOneIteration = Math.min(maxResultsInOneIteration, 1000)
```

Comment:
1000 is too big a number.

Reply:
UFile instances are not that big and 1000 is not that big number it will hardly occupy 1-2MB when loaded into the memory.